worker/uniter/operation/runhook.go

```diff
@@ -195,10 +198,14 @@ func (rh *runHook) afterHook(state State) (_ bool, err error) {
 		if !isLeader || err != nil {
 			return hasRunStatusSet && err == nil, err
 		}
-		rel, err := ctx.Relation(rh.info.RelationId)
-		if err == nil && rel.Suspended() {
+		rel, rErr := ctx.Relation(rh.info.RelationId)
+		if rErr == nil && rel.Suspended() {
 			err = rel.SetStatus(relation.Suspended)
 		}
+	case hooks.PreSeriesUpgrade:
+		logger.Debugf("completing pre upgrade series hook. updating state of series upgrade.")
+		err = rh.callbacks.SetUpgradeSeriesStatus(params.UnitCompleted)
```

Comment:
What happens if the hook returned an error?  Where are we checking for that?

Reply:
It is piped up the stack.

Reply:
At the moment, an error is handled the same generic way with same default generic hook failed callback scenario. I'm not sure if the PR serviced by this story requires that we set the `UnitErrored` status.